Sources/StreamChat/WebSocketClient/EventMiddlewares/UserChannelBanEventsMiddleware.swift

```diff
@@ -18,41 +16,33 @@ struct UserChannelBanEventsMiddleware<ExtraData: ExtraDataTypes>: EventMiddlewar
             completion(event)
             return
         }
-
-        switch event {
-        case let userBannedEvent as UserBannedEvent:
-            database.write { session in
+        
+        do {
+            switch event {
+            case let userBannedEvent as UserBannedEvent:
                 guard let memberDTO = session.member(userId: userBannedEvent.userId, cid: cid) else {
                     throw ClientError.MemberDoesNotExist(userId: userBannedEvent.userId, cid: cid)
                 }
-
+                
                 memberDTO.isBanned = true
                 memberDTO.banExpiresAt = userBannedEvent.expiredAt
-            } completion: { error in
-                if let error = error {
-                    log.error("Failed to write `UserBannedEvent` updates to the DB: \(error)")
-                }
-                completion(event)
-            }
-
-        case let userUnbannedEvent as UserUnbannedEvent:
-            database.write { session in
+                
+            case let userUnbannedEvent as UserUnbannedEvent:
                 guard let memberDTO = session.member(userId: userUnbannedEvent.userId, cid: cid) else {
                     throw ClientError.MemberDoesNotExist(userId: userUnbannedEvent.userId, cid: cid)
                 }
-
+                
                 memberDTO.isBanned = false
                 memberDTO.banExpiresAt = nil
-            } completion: { error in
-                if let error = error {
-                    log.error("Failed to write `UserUnbannedEvent` updates to the DB: \(error)")
-                }
-                completion(event)
+                
+            default:
+                break
             }
-
-        default:
-            completion(event)
+        } catch {
+            log.error("Error handling `\(type(of: event))` event: \(error)")
         }
+        
+        completion(event)
```

Comment:
nit: I really like `defer { completion(event) }` approach used in some middlewares, it's really clear that an event will be forwarded in any case, no matter if the middleware is able to process it. Maybe we can update all the middlewares to follow this pattern (not necessarily in this PR)?